lib/ansible/utils/collection_loader/_collection_finder.py

```diff
@@ -233,7 +233,7 @@ def __init__(self, paths=None, scan_sys_paths=True):
 
             # ensure we always have ansible_collections
             if os.path.basename(p) == 'ansible_collections':
-                p = os.path.dirname(p)
+                p = os.path.realpath.dirname(p)
```

Comment:
not really the correct fix, as that 'real path' might not match the configured paths, keep track of real path and skip dupes